pkg/daemons/control/server.go

```diff
@@ -116,6 +118,7 @@ func controllerManager(ctx context.Context, cfg *config.Control) error {
 		"cluster-cidr":                     util.JoinIPNets(cfg.ClusterIPRanges),
 		"root-ca-file":                     runtime.ServerCA,
 		"profiling":                        "false",
+		"cert-dir":                         certDir,
```

Comment:
The upstream docs for scheduler and controller-manager say:
> If HTTPS serving is enabled, and --tls-cert-file and --tls-private-key-file are not provided, a self-signed certificate and key are generated for the public address and saved to the directory specified by --cert-dir.

So I think that we should be setting --tls-cert-file and --tls-private-key-file, NOT --cert-dir, to ensure that these files do not get overwritten.